File: foodloop_mobile/lib/features/auth/services/auth_service.dart
import 'dart:convert';
import 'dart:developer';
import 'package:foodloop_mobile/core/constants/api_constants.dart';
import 'package:http/http.dart' as http;
import 'package:shared_preferences/shared_preferences.dart';

class AuthService {
  
  Future<bool> login(String email, String password) async {
    try {
      final response = await http.post(
        Uri.parse(ApiConstants.login),
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode({'email': email, 'password': password}),
      );
      print(response.body);
      // if (response.statusCode == 200) {
        final data = jsonDecode(response.body);
        // Store the token
        final prefs = await SharedPreferences.getInstance();
        await prefs.setString('authToken', data['token']);
        return true;
      // }
      // return false;
    } catch (e) {
      
      print('Login error: $e');
      return false;
    }
  }
  
  Future<bool> signup(Map<String, dynamic> userData) async {
    try {
      final response = await http.post(
        Uri.parse(ApiConstants.signUp),
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode(userData),
      );
      print(response.body);
      return response.statusCode == 200;
    } catch (e) {
      print('Signup error: $e');
      return false;
    }
  }
  
  Future<String?> getAuthToken() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString('authToken');
  }
  
  Future<void> logout() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove('authToken');
  }
  
  Future<bool> isLoggedIn() async {
    final token = await getAuthToken();
    log('token: $token');
    return token != null;
  }
  
  Future<Map<String, dynamic>> getUserProfile() async {
    final token = await getAuthToken();
    final response = await http.get(
      Uri.parse(ApiConstants.userProfile),
      headers: {
        'Authorization': 'Bearer $token',
        'Content-Type': 'application/json',
      },
    );
    log('response: ${response.body}');
    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    }
    return {};
  }

  Future<String?> getUserId() async {
    final profile = await getUserProfile();
    return profile.containsKey('_id') ? profile['_id'] : null;
  }
}

File: client/src/services/authService.js
const API_URL = `${import.meta.env.VITE_BACKEND_API}/api/auth`;

export const signupUser = async (userData) => {
  try {
    console.log("Signup request data:", userData);
 

    const response = await fetch(`${API_URL}/signup`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(userData),
    });
    return response.json();
  } catch (error) {
    console.error("Signup error:", error);
    return { error: "Signup failed" };
  }
};

export const loginUser = async (email, password) => {
  try {
    const response = await fetch(`${API_URL}/login`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ email, password }),
    });
    const data = await response.json();

    if (data.token) {
      localStorage.setItem("token", data.token); // Store JWT token
    }

    return data;
  } catch (error) {
    console.error("Login error:", error);
    return { error: "Login failed" };
  }
};

export const logoutUser = async () => {
  localStorage.removeItem("token"); // Remove JWT on logout
  return true;
};
